### PR TITLE
Fixes stormpath/stormpath-sdk-java#531

### PR DESCRIPTION
**Pull request by jarias**
Fixes stormpath/stormpath-sdk-java#531

Update Product Guide to replace

`client.instantiate(AccountStoreMapping.class)`

@mrioan the original PR got close cause I deleted the branch

### CODE REVIEW

File: source/java/product-guide/index.markdown [hunk 1]
```diff
@@ -1611,7 +1611,7 @@ _Account Store_ is a generic term for either a [Directory](#directories) or a [G
 In Stormpath, you control who may login to an application by associating (or 'mapping') one or more account
 stores to an application.  All of the accounts across all of an application's assigned account stores form the application's effective _user base_; those accounts may login to the application.  If no account stores are assigned to an application, no accounts will be able to login to the application.
 
-You control which account stores are assigned (mapped) to an application, and the order in which they are consulted during a login attempt, by manipulating an application's `AccountStoreMapping` resources.
+You control which account stores are assigned (mapped) to an application, and the order in which they are consulted during a login attempt, by manipulating an application's `ApplicationAccountStoreMapping` resources.
```
Comment (mrioan) [inline source/java/product-guide/index.markdown]: last word should be singular: resource.

Comment (jarias) [inline source/java/product-guide/index.markdown]: But is talking about the collection of mappings 

> You control which account stores are assigned

File: source/java/product-guide/index.markdown [hunk 2]
```diff
@@ -1743,8 +1743,8 @@ As we've [shown previously](#workflow-login-attempt), an account trying to login
 
 If you wish to change an account store's login priority for an application, you simply:
 
-1. Find the `accountStoreMapping` resource in the application's `accountStoreMappings` [collection](#collections) that reflects the `accountStore` that you wish to re-prioritize.
-2. Issue an update request to that `AccountStoreMapping`'s instance with a new `listIndex` value.
+1. Find the `applicationAccountStoreMapping` resource in the application's `accountStoreMappings` [collection](#collections) that reflects the `accountStore` that you wish to re-prioritize.
+2. Issue an update request to that `ApplicationAccountStoreMapping`'s instance with a new `listIndex` value.
```
Comment (mrioan) [inline source/java/product-guide/index.markdown]: down here it says "by mapping `$accountStoreMapping` has". Could you please update it to "by mapping `accountStoreMapping` has". (note that I simply deleted the $))

File: source/java/product-guide/index.markdown [hunk 3]
```diff
@@ -1757,14 +1757,14 @@ For example, assume that an account store represented by mapping `$accountStoreM
 
 Applications cannot store Accounts directly - Accounts are always stored in an Account Store (a Directory or Group). Therefore, if you would like an application to be able to create new accounts, you must specify which of the application's associated account stores should store the application's newly created accounts.  This designated account store is called the application's _default account store_.
 
-You specify an application's default account store by setting the AccountStoreMapping's `isDefaultAccountStore` attribute to equal `true`.  You can do this when you create the `accountStoreMapping` resource.  Or if the resource has already been created:
+You specify an application's default account store by setting the ApplicationAccountStoreMapping's `isDefaultAccountStore` attribute to equal `true`.  You can do this when you create the `applicationAccountStoreMapping` resource.  Or if the resource has already been created:
```
Comment (mrioan) [inline source/java/product-guide/index.markdown]: Remove "equal" in "...attribute to equal `true`...". Therefore it should be like this "...attribute to `true`...."

File: source/java/product-guide/index.markdown [hunk 4]
```diff
@@ -1757,14 +1757,14 @@ For example, assume that an account store represented by mapping `$accountStoreM
...
-2. Issue an update request to that AccountStoreMapping's instance with `defaultAccountStore` set to `true`.
+1. Find the `applicationAccountStoreMapping` resource in the Application's `accountStoreMappings` [collection](#collections) that reflects the `accountStore` you wish to be the application's default account store.
+2. Issue an update request to that ApplicationAccountStoreMapping's instance with `defaultAccountStore` set to `true`.
 
 **Example Request**
 
-    accountStoreMapping.setDefaultAccountStore(Boolean.TRUE).save();
+    applicationAccountStoreMapping.setDefaultAccountStore(Boolean.TRUE).save();
 
 Now, any time a new account is created from an application's `createAccount` method, the account will actually be created in the designated default account store.
```
Comment (mrioan) [inline source/java/product-guide/index.markdown]: In the "List Account Store Mappings" section it should say:

The response is a paginated list of `ApplicationAccountStoreMapping` resources.  You may use collection [pagination](#pagination) query parameters to customize the paginated response.
 
 **Example Request**
 
     ApplicationAccountStoreMappingList accountStoreMappings = application.getAccountStoreMappings();
     for (ApplicationAccountStoreMapping accountStoreMapping : accountStoreMappings) {
     	System.out.println(accountStoreMapping.getAccountStore().getHref());
     }

**Comment by mrioan** [comment]
@jarias I made a few comments. If you agree with all of them you can just consider this PR approved. Thanks. Back to you @jarias

File: source/java/product-guide/index.markdown (hunk 1 above)
Comment (mrioan) [inline source/java/product-guide/index.markdown]: ok, then it should be "by manipulating the application's `ApplicationAccountStoreMapping` resources." ("an" replaced with "the"). Makes sense this way?

Comment (jarias) [inline source/java/product-guide/index.markdown]: Yeah that does make sense

**Pull request merged**